src/NCronJob/Registry/JobDefinition.cs

```diff
@@ -82,13 +89,14 @@ public static JobDefinition CreateTyped(
             throw new InvalidOperationException($"Type '{type}' doesn't implement '{nameof(IJob)}'.");
         }
 
-        return new(type, parameter);
+        return new(name, type, parameter);
     }
 
     public static JobDefinition CreateUntyped(
+        string? name,
```

Comment:
Given how often this is `null`, either we push it to the end as an optional argument where `null` is the default or create two methods: `CreateUntpyed(string hash, JEA meta)` and `CreateNamedUntyped(string name, string ...)`

Reply:
Good catch. I'll add an overload.

Reply:
Fixed